### PR TITLE
Rename "statusCodeMatch" to "getTransformFunction"

### PR DESCRIPTION
**Pull request by iamigo**
First look for a match in transform.errorHandlers. If found, use it. If not found, if status code is 2xx, use transform.default, else return false.

### CODE REVIEW

File: src/RefocusCollectorEval.js [hunk 1]
```diff
@@ -90,28 +89,49 @@ class RefocusCollectorEval {
...
+    /*
+     * First check for a status match in transform.errorHandlers (including any
+     * "override" for any 2xx status the template may define).
+     */
+    if (transform.errorHandlers) {
+      /*
+       * Sort the errorHandlers keys alphabetically so there's a predictable
+       * order of evaluation, since we return the "first" match.
+       */
+      const statusMatchers = Object.keys(transform.errorHandlers).sort();
```
Comment (jgraff2) [inline src/RefocusCollectorEval.js]: I think we should maintain the order its defined in. So I can do
'404': () => {},
'4..': () => {},
and be sure that a 404 will be handled by the first one.

Comment (iamigo) [reply src/RefocusCollectorEval.js]: Object.keys doesn't necessarily return the keys in the order in which they are defined:

console.log(Object.keys({ 404: '404', '4..': '4..' }))
// ["404", "4.."]

console.log(Object.keys({ '4..': '4..', '404': '404' }))
// ["404", "4.."]

console.log(Object.keys({ '4..': '4..', 500: '500', '404': '404' }))
// ["404", "500", "4.."]

console.log(Object.keys({ 500: '500', '4..': '4..', '404': '404' }))
// ["404", "500", "4.."]

Comment (jgraff2) [reply src/RefocusCollectorEval.js]: Good point. Ideally we would be able to support the case above though. Not sure if there's a good way to do that. Thoughts?

Maybe we could match all the \d\d\d ones first, then match the others if necessary?

Comment (iamigo) [reply src/RefocusCollectorEval.js]: I like the idea of trying to do exact matches on \d\d\d first -- I'll do that.

File: test/getTransformFunction.js [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+    } catch (err) {
+      done(err);
+    }
+  });
+
+  it('error handler match - 500', (done) => {
+    try {
+      collectRes.res.statusCode = 500;
+      const expectedRetVal = collectRes.generatorTemplate.transform
+        .errorHandlers['5..'];
```
Comment (jgraff2) [inline test/getTransformFunction.js]: this should match to errorHandlers['500'] now, right?

File: test/getTransformFunction.js [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+      const retval = getTransformFunction(collectRes.generatorTemplate
+        .transform, collectRes.res.statusCode);
+      expect(retval).to.equal(false);
+      done();
+    } catch (err) {
+      done(err);
+    }
+  });
+
+  it('error handler multiple match - should return first match', (done) => {
```
Comment (jgraff2) [inline test/getTransformFunction.js]: This is the same as the test on line 100.

**Comment by iamigo** [comment]
Fixed per Jon's comments.

**Pull request merged**